graylog2-server/src/main/java/org/graylog2/migrations/MaintenanceStrategiesHelper.java

```diff
@@ -78,16 +69,13 @@ public ImmutablePair<String, RotationStrategyConfig> readRotationConfigFromServe
             }
             case TimeBasedSizeOptimizingStrategy.NAME -> {
                 return ImmutablePair.of(TimeBasedSizeOptimizingStrategy.class.getCanonicalName(),
-                        TimeBasedSizeOptimizingStrategyConfig.builder()
-                                .indexLifetimeMin(elasticsearchConfiguration.getTimeSizeOptimizingRotationMinLifeTime())
-                                .indexLifetimeMax(elasticsearchConfiguration.getTimeSizeOptimizingRotationMaxLifeTime())
-                                .build());
+                        buildDefaultTimeSizeStrategy());
             }
             default -> {
-                LOG.warn("Unknown retention strategy [{}]. Defaulting to [{}]",
-                        elasticsearchConfiguration.getRotationStrategy(), MessageCountRotationStrategy.NAME);
-                return ImmutablePair.of(MessageCountRotationStrategy.class.getCanonicalName(),
-                        MessageCountRotationStrategyConfig.create(elasticsearchConfiguration.getMaxDocsPerIndex()));
+                LOG.warn("Unknown rotation strategy [{}]. Defaulting to [{}].",
+                        elasticsearchConfiguration.getRotationStrategy(), TimeBasedSizeOptimizingStrategy.NAME);
+                return ImmutablePair.of(TimeBasedSizeOptimizingStrategy.class.getCanonicalName(),
+                        buildDefaultTimeSizeStrategy());
```

Comment:
This changes the fall-back for unknown/invalid rotation strategies in the `server.conf` to use the new Time Size Optimizing strategy.